internal/provider/resource_tfe_organization_run_task.go

```diff
@@ -1,209 +1,272 @@
 // Copyright (c) HashiCorp, Inc.
 // SPDX-License-Identifier: MPL-2.0
 
-// NOTE: This is a legacy resource and should be migrated to the Plugin
-// Framework if substantial modifications are planned. See
-// docs/new-resources.md if planning to use this code as boilerplate for
-// a new resource.
-
 package provider
 
 import (
 	"context"
+	"errors"
 	"fmt"
-	"log"
-	"strings"
 
 	tfe "github.com/hashicorp/go-tfe"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	customValidators "github.com/hashicorp/terraform-provider-tfe/internal/provider/validators"
 )
 
-func resourceTFEOrganizationRunTask() *schema.Resource {
-	return &schema.Resource{
-		Create: resourceTFEOrganizationRunTaskCreate,
-		Read:   resourceTFEOrganizationRunTaskRead,
-		Delete: resourceTFEOrganizationRunTaskDelete,
-		Update: resourceTFEOrganizationRunTaskUpdate,
-		Importer: &schema.ResourceImporter{
-			StateContext: resourceTFEOrganizationRunTaskImporter,
-		},
+type resourceOrgRunTask struct {
+	config ConfiguredClient
+}
+
+func NewOrganizationRunTaskResource() resource.Resource {
+	return &resourceOrgRunTask{}
+}
+
+type modelTFEOrganizationRunTaskV0 struct {
+	Category     types.String `tfsdk:"category"`
+	Description  types.String `tfsdk:"description"`
+	Enabled      types.Bool   `tfsdk:"enabled"`
+	HMACKkey     types.String `tfsdk:"hmac_key"`
+	ID           types.String `tfsdk:"id"`
+	Name         types.String `tfsdk:"name"`
+	Organization types.String `tfsdk:"organization"`
+	URL          types.String `tfsdk:"url"`
+}
 
-		CustomizeDiff: customizeDiffIfProviderDefaultOrganizationChanged,
```

Comment:
There is a framework replacement for this CustomizeDiff. You have to implement `ResourceWithModifyPlan` on the main type. Here's [an example](https://github.com/hashicorp/terraform-provider-tfe/blob/364f05a731f9f97ef21534fa978d91c3fbcc182c/internal/provider/resource_tfe_registry_gpg_key.go#L41-L43)

Without it, changes to the provider default organization are not detected on this resource. You can try it yourself by using a default in the provider config and then changing it:

```
provider "tfe" {
  organization = "hashicorp" # <-- apply this, then change it. No changes are planned for tfe_organization_run_task
}

resource "tfe_workspace" "example" {
  name = "example-run-task"
}

resource "tfe_organization_run_task" "example" {
  url          = "https://en2obyzxxt083.x.pipedream.net/"
  name         = "task-name"
  enabled      = true
}
```

Reply:
Fixed this.  Thanks for the find!